lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx

```diff
@@ -32,262 +29,261 @@ describe('NxCollapsibleItems', function() {
 
-    it('fires the components onToggleCollapse when clicked', function() {
-      const onToggleCollapse = jest.fn(),
-          trigger = getShallowTrigger({ onToggleCollapse });
-
-      expect(onToggleCollapse).not.toHaveBeenCalled();
-      trigger.simulate('click');
-      expect(onToggleCollapse).toHaveBeenCalled();
+    it('fires the component\'s onToggleCollapse when clicked if there are children', async function() {
+      const user = userEvent.setup(),
+          onToggleCollapseNoChildren = jest.fn(),
+          onToggleCollapseChildren = jest.fn(),
+          noChildrenTrigger = renderAndGetTrigger({
+            onToggleCollapse: onToggleCollapseNoChildren
+          }),
+          childrenTrigger = renderAndGetTrigger({
+            onToggleCollapse: onToggleCollapseChildren,
+            children: <span />
+          });
+
+      expect(onToggleCollapseNoChildren).not.toHaveBeenCalled();
+      expect(onToggleCollapseChildren).not.toHaveBeenCalled();
+
+      await user.click(noChildrenTrigger);
+      expect(onToggleCollapseNoChildren).not.toHaveBeenCalled();
+
+      await user.click(childrenTrigger);
+      expect(onToggleCollapseChildren).toHaveBeenCalled();
     });
 
-    it('contains an nx-collapsible-items__twisty icon', function() {
-      expect(getShallowTrigger().find(NxFontAwesomeIcon)).toHaveClassName('nx-collapsible-items__twisty');
-    });
+    it('renders the triggerContent', function() {
+      const trigger = renderAndGetTrigger(),
+          triggerContent = screen.getByTestId('trigger-content');
 
-    it('renders the triggerContent as children of a nx-collapsible-items__text element', function() {
-      expect(getShallowTrigger().find('.nx-collapsible-items__text')).toContainReact(minimalProps.triggerContent);
+      expect(trigger).toContainElement(triggerContent);
     });
 
-    it('wraps the trigger in a tooltip with the triggerTooltip object if specified', function() {
-      const component = getShallowComponent({ triggerTooltip: { title: 'tip', placement: 'right' } }),
-          tooltip = component.find(NxTooltip),
-          trigger = tooltip.find('.nx-collapsible-items__trigger');
+    it('sets a tooltip on the trigger as specified by the triggerTooltip object when children exist', async function() {
+      const user = userEvent.setup(),
+          trigger = renderAndGetTrigger({
+            triggerTooltip: { title: 'tip', className: 'foo' },
+            children: <span />
+          });
+
+      await runTimers();
+      expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+      await user.hover(trigger);
+      await runTimers();
 
-      expect(tooltip).toExist();
-      expect(tooltip).toHaveProp('title', 'tip');
-      expect(tooltip).toHaveProp('placement', 'right');
+      const tooltip = screen.getByRole('tooltip');
 
-      expect(trigger).toExist();
+      expect(tooltip).toHaveTextContent('tip');
+      expect(tooltip.querySelector('.foo')).toBeInTheDocument();
```

Comment:
Is this to test if the classname is set correctly?

Reply:
> Is this to test if the classname is set correctly?

Yes but we don't have control over exactly which element the class gets set on, it's up to how MUI builds the tooltip.  So I can't assert that the class is on the tooltip, but I can assert that it's on some descendant of the tooltip.

Reply:
Make sense, thanks.